src/Silex/Provider/TranslationServiceProvider.php

```diff
@@ -80,9 +80,9 @@ public function register(Container $app)
             return new MessageSelector();
         };
 
-        $app['translator.resources'] = $app->protect(function ($app) {
+        $app['translator.resources'] = function ($app) {
             return array();
-        });
+        };
```

Comment:
This looks perfectly valid to me. And extended a protected closure worked before. So, the fix on Pimple looks wrong IMHO.